packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -394,9 +403,18 @@ class _ModalScopeState extends State<_ModalScope> {
   void dispose() {
     config.route.animation?.removeStatusListener(_animationStatusChanged);
     config.route.forwardAnimation?.removeStatusListener(_animationStatusChanged);
+    willPopCallbacks = const <WillPopCallback>[];
     super.dispose();
   }
 
+  void addWillPopCallback(WillPopCallback callback) {
```

Comment:
should probably warn that the callback might be called long after the route was unmounted.